providers/cncf/kubernetes/src/airflow/providers/cncf/kubernetes/utils/pod_manager.py

```diff
@@ -865,8 +865,7 @@ def _exec_pod_command(self, resp, command: str) -> str | None:
         self.log.info("Running command... %s", command)
         resp.write_stdin(f"{command}\n")
         while resp.is_open():
-            resp.update(timeout=1)
-            while resp.peek_stdout():
```

Comment:
(I'll preface this by saying I don't have hands on experience here)

Are you sure this is the right approach? Is there anything to peek if we don't update? I wonder if the real fix is not to return after reading the first chunk?

This is probably worth adding a test to cover this, maybe an integration test. If nothing else, that'll make me more confident this is the right fix.